sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-files/ece-response-file-detail.ts

```diff
@@ -73,7 +73,7 @@ export class ECEResponseFileDetail extends ECEResponseFileRecord {
     if (isNaN(this.disbursementIdentifier)) {
       errors.push("Invalid unique index number for the disbursement record");
     }
-    if (!this.applicationNumber?.trim()) {
+    if (!this.applicationNumber?.trim() || isNaN(+this.applicationNumber)) {
```

Comment:
do we need `+` when using `isNaN` `?

Reply:
if it only excepts then what is the use of not a number check, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN, 
![image](https://github.com/bcgov/SIMS/assets/77353155/7b3e23e7-37e0-419f-a6f5-b270e1e2bdf4)

Reply:
Yes, that's weird but in TS that's the way. I think it still checks for null/undefined.
https://stackoverflow.com/a/42121162/7983709

Reply:
ya @andrepestana-aot. it says if we use `as any` then it bypasses compiler check and works as JS, weird
![image](https://github.com/bcgov/SIMS/assets/77353155/fa7cdb5c-c893-4095-9b30-dbff34a1b6c3)